out/notebooks/splunk.ts

```diff
@@ -29,95 +30,157 @@ export function splunkLogin(service) {
         
         service.login(function(err, wasSuccessful) {
             if (err !== null || !wasSuccessful) {
-                reject(err)
+                reject(err);
             } else {
-                resolve(null)
+                resolve(null);
             }
-        })
+        });
 
-    })
+    });
 
 
 }
 
 
 export function createSearchJob(jobs, query, options) {
-
     return new Promise(function(resolve, reject) {
         jobs.create(query, options, function(err, data) {
             if (err !== null) {
                 reject(err);
             } else {
-                resolve(data)
+                resolve(data);
             }
-        })
+        });
+
+    });
+}
 
-    })
+export function dispatchSpl2Module(service: any, spl2Module: string) {
+    // Get first statement assignment '$my_statement = ...' -> 'my_statement' 
+    const statementMatch = spl2Module.match(/\$([a-zA-Z0-9_]+)[\s]*=/m);
```

Comment:
I'm not sure how existing ICX frontends parse the SPL2 looking for statements but this is some simple regex to get us started. It probably makes sense to mimic what other frontends are doing here though.